tests/test_searcher.py

```diff
@@ -85,63 +93,115 @@ def _run_perf_analytics(
+        import os
+        import sys
+
+        test_name = (
+            test_name.replace("[", "_")
+            .replace("]", "_")
+            .replace(" ", "_")
+            .replace("/", "_")
+        )
+        perf_test_folder = "perf/"
+
+        if not os.path.exists(perf_test_folder):
+            os.mkdir(perf_test_folder)
+
+        with open(
+            os.path.join(perf_test_folder, f"{test_name}.txt"),
+            "w",
+        ) as f:
+            sys.stdout = f
+            print(
+                "------------------------------------------------------------------------------------------------"
+            )
+            print(f"FEN: {fen}")
+            stats = pstats.Stats(profiler)
+            stats.strip_dirs().sort_stats("tottime").print_stats()
+            print(
+                "------------------------------------------------------------------------------------------------"
+            )
+
+        sys.stdout = sys.__stdout__
 
     @pytest.mark.slow
-    def test_perf_base(self, fen_string: str, max_depth: int) -> None:
-        """Performance test base"""
+    def test_perf_base(self, request_fixture, fen_string: str, max_depth: int) -> None:
         self._run_perf_analytics(
+            request_fixture.node.name,
             fen=fen_string,
             max_depth=max_depth,
         )
 
     @pytest.mark.slow
-    def test_perf_transposition_table(self, fen_string: str, max_depth: int) -> None:
-        """Performance test with transposition table"""
+    def test_perf_transposition_table(
+        self, request_fixture, fen_string: str, max_depth: int
+    ) -> None:
         self._run_perf_analytics(
+            request_fixture.node.name,
             fen=fen_string,
             max_depth=max_depth,
             enable_transposition_table=True,
         )
 
     @pytest.mark.slow
-    def test_perf_null_move_pruning(self, fen_string: str, max_depth: int) -> None:
-        """Performance test with null move pruning"""
+    def test_perf_null_move_pruning(
+        self, request_fixture, fen_string: str, max_depth: int
+    ) -> None:
         self._run_perf_analytics(
+            request_fixture.node.name,
             fen=fen_string,
             max_depth=max_depth,
             enable_null_move_pruning=True,
         )
 
     @pytest.mark.slow
-    def test_perf_aspiration_windows(self, fen_string: str, max_depth: int) -> None:
-        """Performance test with aspiration windows"""
+    def test_perf_aspiration_windows(
+        self, request_fixture, fen_string: str, max_depth: int
+    ) -> None:
         self._run_perf_analytics(
+            request_fixture.node.name,
             fen=fen_string,
             max_depth=max_depth,
             enable_aspiration_windows=True,
         )
 
     @pytest.mark.slow
-    def test_perf_delta_pruning(self, fen_string: str, max_depth: int) -> None:
-        """Performance test with aspiration windows"""
+    def test_perf_futility_pruning(
```

Comment:
Dont see any test for `futility_pruning` 👀

Reply:
I actually did try before but the results came back different. I'm not sure we actually expect it to be consistent in every case

Reply:
Agree that consistency test is not really suitable for the one. Maybe just add a line somewhere checking computation logic?

Reply:
What logic do you suggest to test? I don't think it will be easy in any case. Statistics aren't propagated back.

Reply:
Good argument. Happy to let go (for now). 

As you know, I really like tests - maybe assigning myself a task just cleaning up tests in general. Happy to work on this this coming week.

Reply:
I'll add one once I propagate stats, this is needed - agreed

Reply:
tyty!